checkov/terraform/checks/provider/base_check.py

```diff
@@ -11,9 +12,19 @@ def __init__(self, name, id, categories, supported_provider):
         self.supported_provider = supported_provider
         provider_registry.register(self)
 
+    def scan_entity_conf(self, conf, entity_type):
+        return self.scan_provider_conf(conf, entity_type)
+
     @abstractmethod
-    def scan_provider_conf(self, conf):
+    @multi_signature()
+    def scan_provider_conf(self, conf, entity_type):
         raise NotImplementedError()
 
-    def scan_entity_conf(self, conf):
-        return self.scan_provider_conf(conf)
+    @classmethod
+    @scan_provider_conf.add_signature(args=["self", "conf"])
+    def _scan_provider_conf_self_conf(cls, wrapped):
+        def wrapper(self, conf, entity_type=None):
+            # keep default argument for entity_type so old code, that doesn't set it, will work.
+            return wrapped(self, conf)
```

Comment:
@angatha Wouldn't the `entity_type` be always `provider` here?

Reply:
Yes

Reply:
@nimrodkor In this case (and the one below) the `entity_type` add no value and it could be removed.

Reply:
@nimrodkor this was not correct. In the case of provider, this is the name of the provider. `tests.terraform.runner.TestRunnerValid.test_runner_two_checks_only` will fail, if I would have added an assert statement.